icefoganalytics/elcc-data-management
# ELCC-64: Creating Worksheets for 2024/25 Does Not Work

Fixes https://yg-hpw.atlassian.net/browse/ELCC-64

# Context

There is a feature to add worksheets for a given year if they don’t exist. This feature does not seem to work for any year except 2023/24. Also the redirect from the Child Care Centres page no longer redirects to the correct year. It redirects to 2024/25 instead of 2023/24.

## Reproduction Instructions

1. Go to http://localhost:8080/child-care-centres.
2. Double click on a child care center or create one.
3. Note that this will set the Fiscal year to 2024/25 instead of 2023/24.
4. Go to the “Worksheets” tab.
5. Try to create worksheets for a year that doesn’t have any via the “add worksheets for xxx” button.
6. This will only work for 2023/24.
7. This might be an issue with missing seeds in Development?

# Implementation

This turned out to be a seeding issue. The seeds only supported worksheet creation on 2023/24.
Also, the current fiscal year function was correct, just so much time has passed that its now 2024/25.

I added some tests for the function and updated the seeds to be more future proof.

# Screenshots

![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/d4306695-920d-4b25-838b-84d6aaee2f78)

# Testing Instructions

1. Boot the app via `dev up`
2. Run the seeds via `dev seed up` - there are a lot more of them now so it takes a while.
3. Check that the app complies, and that you can log in at http://localhost:8080.
4. Go to http://localhost:8080/child-care-centres.
5. Double click on a child care center or create one.
6. Note that this will set the Fiscal year to 2024/25 instead of 2023/24.
7. Go to the “Worksheets” tab.
8. Try to create worksheets for a year that doesn’t have any via the “add worksheets for xxx” button.
9. This will now work for 2023/24.

> If you already created the worksheets, you will need to delete them via the database console. We currently don't have a way to "regenerate" the worksheets.
